api/rest/restapi.go

```diff
@@ -970,6 +1008,26 @@ func (api *API) parseCidOrError(w http.ResponseWriter, r *http.Request) types.Pi
 	return pin
 }
 
+func (api *API) parsePath(w http.ResponseWriter, r *http.Request) (string, types.PinSerial) {
```

Comment:
if we do `path: .*` I think this method is not needed.

Reply:
The value of `path` in `/pins/{path: .*}` would not start with `/`
paths like `ipfs/Q....`, `ipns/Q..` are not valid paths(ParsePath would fail). A cid starting without slash is valid path, but if it starts with slash, it is not.
I think we would need logic to add slashes for ipfs, ipns, ipld before we send paths for `Resolve`
https://github.com/ipfs/go-path/blob/1bd072233a4bb2c9d6aabceea7cd16ea3cd23973/path.go#L92-L140

Reply:
Ah right. Yes we'll need that.